### PR TITLE
Send new BATTERY_EXTENDED_STATUS message

### PR DESCRIPTION
**Pull request by peterbarker**
Supposed to contain lower-rate data.

Alternative to https://github.com/ArduPilot/ardupilot/pull/17869 which @WickedShell has reservations about due to bandwidth issues.

### CODE REVIEW

**Comment by peterbarker** [comment]
```
STABILIZE> 976: BATTERY_EXTENDED_STATUS {id : 0, internal_resistance : 0.000883447122760117}
```

**Comment by tridge** [comment]
using a new OPTIONS bit per battery to ask it to send the resting voltage estimate over mavlink would be simpler in many ways

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -284,6 +284,35 @@ bool GCS_MAVLINK::send_battery_status()
...
+{
+    const AP_BattMonitor &battery = AP::battery();
+
+    for(uint8_t i = 0; i < AP_BATT_MONITOR_MAX_INSTANCES; i++) {
+        const uint8_t battery_id = (last_battery_extended_status_idx + 1) % AP_BATT_MONITOR_MAX_INSTANCES;
+        if (battery.get_type(battery_id) == AP_BattMonitor::Type::NONE) {
+            last_battery_extended_status_idx = battery_id;
+            continue;
+        }
+        CHECK_PAYLOAD_SIZE(BATTERY_EXTENDED_STATUS);
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp:308]: only for batteries that can monitor both voltage and current

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -284,6 +284,35 @@ bool GCS_MAVLINK::send_battery_status()
...
+        send_battery_extended_status(battery_id);
+        last_battery_extended_status_idx = battery_id;
+        return true;
```
Comment (nkruzan) [inline libraries/GCS_MAVLink/GCS_Common.cpp:311]: doesn't this only report the first battery? comment is "returns true if all battery instances were reported" but seems to return true after first message sent.

**Pull request closed without merging**